### DISCUSSION TITLE
What exactly does the --dir (preopened directory) flag do? How does it differ from --mapdir?

### DISCUSSION THREAD

**Discussion by artlogic** [wasmerio/wasmer#2532] [Q&A]
The subject says it all. I'm trying to understand what `--dir` does, and why it exists? How does it work with `--mapdir`? I've seen some odd behavior when using these with the WASI SDK and I'd like to understand what `--dir`'s intention is. Thanks1

**Comment by syrusakbary** [comment]
`--dir` makes accessible certain filesystem directory to the program. Without it, the WASI program is not able to access the outside world (that means, that the programs are sandboxed by default)

**Reply by artlogic** [reply]
> `--dir` makes accessible certain filesystem directory to the program. Without it, the WASI program is not able to access the outside world (that means, that the programs are sandboxed by default)

How does that differ from `--mapdir`?
